hoomd/SnapshotSystemData.cc

```diff
@@ -39,6 +39,24 @@ void SnapshotSystemData<Real>::replicate(unsigned int nx, unsigned int ny, unsig
     pair_data.replicate(n, old_n);
     }
 
+template<class Real> void SnapshotSystemData<Real>::wrap()
+    {
+    for (unsigned int i = 0; i < particle_data.size; i++)
+        {
+        auto const frac = global_box.makeFraction(particle_data.pos[i]);
+        auto modulus_positive
+            = [](Real x) { return std::fmod(std::fmod(x, Real(1.0)) + Real(1.0), Real(1.0)); };
+        auto const wrapped = vec3<Real>(modulus_positive(static_cast<Real>(frac.x)),
+                                        modulus_positive(static_cast<Real>(frac.y)),
+                                        modulus_positive(static_cast<Real>(frac.z)));
+        particle_data.pos[i] = global_box.makeCoordinates(wrapped);
+        auto const img = make_int3(static_cast<int>(std::floor(frac.x)),
+                                   static_cast<int>(std::floor(frac.y)),
+                                   static_cast<int>(std::floor(frac.z)));
+        particle_data.image[i] = img;
```

Comment:
Do we need to consider wrapping particles that already have a nonzero image, and handling the image as an offset? I would expect that the `wrap` method is idempotent for particles inside the box, but this would reset their images to [0, 0, 0] instead of keeping the existing image flags.

This case should be tested, if you agree.
```suggestion
        particle_data.image[i] += img;
```

Reply:
Yes, this would be good to add and test.

Reply:
Good catch, will implement this weekend